### PR TITLE
feat(swift): Support for Package.swift files

### LINKED ISSUE #3880: Support Swift Package Manager
https://github.com/apple/swift-package-manager/blob/master/Documentation/PackageDescription.md#package-dependency

### PR DESCRIPTION
**Pull request by zharinov**
Closes #3880

### CODE REVIEW

**Comment by zharinov** [comment]
https://github.com/zharinov/vapor/pulls

**Comment by rarkins** [comment]
@zharinov we recently introduced the concept of `displayFrom` and `displayTo` in the code. If you search for where it's used already, perhaps you can work out how to pretty up the PR titles for Swift too?

**Comment by zharinov** [comment]
> @zharinov we recently introduced the concept of `displayFrom` and `displayTo` in the code. If you search for where it's used already, perhaps you can work out how to pretty up the PR titles for Swift too?

I see, but unfortunately found no good place to `displayTo` field. Another thing to consider is it looks ok in the body of PR, so only title need to be changed.

**Comment by zharinov** [comment]
I think I found some tradeoff 
image

**Comment by rarkins** [comment]
I'll have a play with it. I'd like to remove the "from: " from all places and just use versions.

**Comment by zharinov** [comment]
In this case we'll lose its range semantics (`from: 1.2.3` stands for `>=1.2.3 <2.0.0`), but it may be okay

**Comment by rarkins** [comment]
OK that's interesting. I guess it's a tradeoff then. So "1.2.3" means exactly 1.2.3 while "from: 1.2.3" means the range?

**Comment by zharinov** [comment]
Yes, it's can be written like `.exact("1.2.3")` and `"1.2.3"..<"2.0.0"` respectively, in Swift terms. That's whay I've made its own `versionScheme`.

File: lib/config/definitions.js [hunk 1]
```diff
@@ -1811,6 +1812,19 @@ const options = [
...
   },
+  {
+    name: 'swift',
+    description: 'Configuration for Package.swift files',
+    stage: 'package',
+    type: 'object',
+    default: {
+      fileMatch: ['(^|/)Package\\.swift'],
+      versionScheme: 'swift',
+      rangeStrategy: 'bump',
```
Comment (rarkins) [inline lib/config/definitions.js]: Why do you suggest `bump` by default?

File: lib/manager/swift/extract.js [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+  } catch (e) {
+    return null;
+  }
+}
+
+function extractPackageFile(content, packageFile = null) {
+  if (!content) return null;
+
+  const result = {
+    packageFile,
```
Comment (rarkins) [inline lib/manager/swift/extract.js]: I don't think we want to set this if `packageFile` is null. Or will null only happen in testing?

File: lib/datasource/git-tags/index.js [hunk 3]
```diff
@@ -3,39 +3,50 @@ const { getRemoteInfo } = require('isomorphic-git');
 const cacheNamespace = 'git-tags';
 const cacheMinutes = 10;
 
-async function getPkgReleases({ lookupName }) {
+async function getPkgReleases({ lookupName, registryUrls = [] }) {
```
Comment (rarkins) [inline lib/datasource/git-tags/index.js]: Can you describe how these semantics works? e.g. it's possible that a single dependency (`depName`) may have multiple repositories (`registryUrls`) to look it up in? Is this a requirement from Swift?

File: website/docs/configuration-options.md [hunk 4]
```diff
@@ -1080,6 +1080,8 @@ Use this field to suppress various types of warnings and other notifications fro
 
 The above config will suppress the comment which is added to a PR whenever you close a PR unmerged.
 
+## swift
```
Comment (rarkins) [inline website/docs/configuration-options.md]: Do you think there's anything about Swift support that should be described? e.g. our handling of `from:` etc?

File: lib/config/definitions.js (hunk 1 above)
Comment (zharinov) [reply lib/config/definitions.js]: As I can see, `.from(<...>)` expressions mostly used with Swift, so `auto` strategy updates almost nothing in my examples. But maybe that's okay and should be reverted back to default.

File: lib/manager/swift/extract.js (hunk 2 above)
Comment (zharinov) [reply lib/manager/swift/extract.js]: Yes, it should be null only when testing

File: lib/config/definitions.js (hunk 1 above)
Comment (zharinov) [reply lib/config/definitions.js]: Also, current version of `getNewValue` support only `bump` semantics.

Comment (rarkins) [reply lib/config/definitions.js]: Does `.from()` skip major versions as well? or is constrained to the current version?

And is there a lock file we could be bumping instead?

File: lib/datasource/git-tags/index.js (hunk 3 above)
Comment (rarkins) [reply lib/datasource/git-tags/index.js]: What if you instead store publisher/repo into `depName` and the full URL into `lookupName`?

File: lib/config/definitions.js (hunk 1 above)
Comment (zharinov) [reply lib/config/definitions.js]: It seems to be constrained on current major

File: lib/datasource/git-tags/index.js (hunk 3 above)
Comment (zharinov) [reply lib/datasource/git-tags/index.js]: Ah, makes sense. Just followed other datasources code I knew.

**Pull request merged**